docs/source_values.rst

```diff
@@ -178,12 +185,106 @@ In this example, the LED is lit only when the button is not pressed:
+
+.. image:: images/source_values/sin_values.*
+    :align: center
+
+.. literalinclude:: examples/sin_values.py
+
+In this example, all three devices are following the `sine wave`_. The motor
+ramps up from 0 (stopped) to 1 (full speed forwards), then back down to 0 and on
+to -1 (full speed backwards) in a cycle. Similarly, the servo moves from its mid
+point to the right, then towards the left; and the buzzer starts with its mid
+tone, gradually raises its frequency, to its highest tone, then down towards its
+lowest tone. Note that setting :attr:`~SourceMixin.source_delay` will alter
+the speed at which the device iterates through the values.
+
+.. _sine wave: https://en.wikipedia.org/wiki/Sine_wave
+
+Internal devices
+----------------
+
+GPIO Zero also provides several :doc:`internal devices <api_internal>` which
+represent facilities provided by the operating system itself. These can be used
+to react to things like the time of day, or whether a server is available on the
+network. These classes include a :attr:`~ValuesMixin.values` property which can
+be used to feed values into a device's :attr:`~SourceMixin.source`. For example,
+a lamp connected to an :class:`Energenie` socket can be controlled by a
+:class:`TimeOfDay` object so that it is on during the hours of 8am and 8pm:
+
+.. image:: images/source_values/timed_heat_lamp.*
+    :align: center
+
+.. literalinclude:: examples/timed_heat_lamp.py
+
+Using the :class:`DiskUsage` class with :class:`LEDBarGraph` can show your Pi's
+disk usage percentage on a bar graph:
+
+.. image:: images/source_values/disk_usage_bar_graph.*
+    :align: center
+
+.. literalinclude:: examples/disk_usage_bar_graph.py
+
+Demonstrating a garden light system whereby the light comes on if it's dark and
+there's motion is simple enough, but it requires using the
+:func:`~tools.booleanized` source tool to convert the light sensor from a float
+value into a boolean:
+
+.. image:: images/source_values/garden_light.*
+    :align: center
+
+.. literalinclude:: examples/garden_light.py
+
+Composite devices
+-----------------
+
+The :attr:`~Device.value` of a composite device made up of the nested values of
+its devices. For example, the value of a :class:`Robot` object is a 2-tuple
+containing its left and right motor values:
+
+.. code-block:: pycon
+
+    >>> from gpiozero import Robot
+    >>> robot = Robot(left=(14, 15), right=(17, 18))
+    >>> robot.value
+    RobotValue(left_motor=0.0, right_motor=0.0)
+    >>> tuple(robot.value)
+    (0.0, 0.0)
+    >>> robot.forward()
+    >>> tuple(robot.value)
+    (1.0, 1.0)
+    >>> robot.backward()
+    >>> tuple(robot.value)
+    (-1.0, -1.0)
+    >>> robot.value = (1, 1)  # robot is now driven forwards
+
+Use two potentiometers to control the left and right motor speed of a robot:
+
+.. image:: images/source_values/robot_pots_1.*
+    :align: center
+
+.. literalinclude:: examples/robot_pots_1.py
```

Comment:
Given that `left` and `right` are used in the robot's constructor, this might be clearer if you used `left_pot` and `right_pot` variables instead of `left` and `right`? (and also in the next example)

Reply:
Ok